src/components/customer-resource-coverage/customer-resource-coverage.js

```diff
@@ -1,42 +1,46 @@
 import React, { Component } from 'react';
 import { Field, FieldArray, reduxForm } from 'redux-form';
 import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
 import moment from 'moment';
 import isEqual from 'lodash/isEqual';
-import classNames from 'classnames/bind';
 
 import Datepicker from '@folio/stripes-components/lib/Datepicker';
 import Button from '@folio/stripes-components/lib/Button';
 import Icon from '@folio/stripes-components/lib/Icon';
 import IconButton from '@folio/stripes-components/lib/IconButton';
-import CoverageDateList from '../coverage-date-list';
 import KeyValueLabel from '../key-value-label';
+
+import CoverageDateList from '../coverage-date-list';
 import styles from './customer-resource-coverage.css';
+import { formatISODateWithoutTime } from '../utilities';
 
 const cx = classNames.bind(styles);
 
 class CustomerResourceCoverage extends Component {
   static propTypes = {
     initialValues: PropTypes.shape({
-      customCoverages: PropTypes.array,
+      customCoverages: PropTypes.array
     }).isRequired,
+    packageCoverage: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
     onSubmit: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func,
     pristine: PropTypes.bool,
     isPending: PropTypes.bool,
     initialize: PropTypes.func,
-    locale: PropTypes.string // eslint-disable-line react/no-unused-prop-types
+    locale: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
+    intl: PropTypes.object // eslint-disable-line react/no-unused-prop-types
   };
 
   state = {
-    isEditing: false,
+    isEditing: false
   };
 
   componentWillReceiveProps(nextProps) {
     let wasPending = this.props.isPending && !nextProps.isPending;
-    let needsUpdate = !isEqual(this.props.initialValues, nextProps.initialValues);
+    let needsUpdate = !isEqual(this.props.initialValues.customCoverages, nextProps.initialValues.customCoverages);
 
-    if (wasPending || needsUpdate) {
+    if (wasPending && needsUpdate) {
```

Comment:
Is this the right logic now? It's still an `or` the couple of other times this pops up.

Reply:
i'll investigate the logic here as compared to the other places we use this pattern, but iirc one of our pairing sessions included changing this to `||`.  i'll try to suss out why

Reply:
verified: `||` causes that behavior where clicking any button on the form immediately closes it.  `&&` is correct